### PR TITLE
Display helpful message for write permissions denied exception

### PR DESCRIPTION
**Pull request by kamilogorek**
It's not necessarily the best solution that we can have, but it'll do the job for now and clarify the issue for some developers.

Fixes https://github.com/ember-cli/ember-cli/issues/6110

### CODE REVIEW

**Comment by nathanhammond** [comment]
@kamilogorek This appears to have failed legitimately on Windows.

In general I'm in favor of providing this feedback to users but I'm unsure when exactly we should perform this test. @stefanpenner @rwjblue Any thoughts?

**Comment by kamilogorek** [comment]
@nathanhammond that's true, but it's failing in the same way in all recent PRs. An issue is hidden somewhere deeper.

File: blueprints/addon/index.js [hunk 1]
```diff
@@ -161,7 +161,23 @@ function alphabetizeDependencies(contents) {
 }
 
 function writeContentsToFile(contents, fileName) {
-  var packagePath = path.join(this.path, 'files', fileName);
+  var dirPath = path.join(this.path, 'files');
+  var packagePath = path.join(dirPath, fileName);
 
-  fs.writeFileSync(packagePath, stringifyAndNormalize(contents));
+  try {
+    fs.writeFileSync(packagePath, stringifyAndNormalize(contents));
```
Comment (stefanpenner) [inline blueprints/addon/index.js]: @rwjblue is this that file that we accidentally write too in peoples `node_modules` ?

Comment (rwjblue) [inline blueprints/addon/index.js]: Yes, it is that file that we have discussed.  Yes, we should absolutely fix it.

Comment (stefanpenner) [inline blueprints/addon/index.js]: @kamilogorek although you PR improves the error message, I suspect we can get rid of the problem in most cases if we don't incorrectly write to this file in some scenarios.

Comment (kamilogorek) [inline blueprints/addon/index.js]: Most definitely. Do you think those files should be written directly into generated folder or stored in an os temp folder first and then moved after being generated?

**Comment by kamilogorek** [comment]
@Turbo87 API for this module has been completely rewritten and I'd need to get familiar with it from scratch. Content creating and writing is in a different place now, so it cannot be rebased.

**Comment by Turbo87** [comment]
Sorry about that! Not sure why this PR was never merged last year... 😞 

I guess in that case we should probably close the PR and if you feel like porting it over to the new code we can open another PR for it, okay?

**Comment by kamilogorek** [comment]
Sure, no worries! I'll try to tackle this one in my spare time :)

**Pull request closed without merging**